### DISCUSSION TITLE
nerdctl save format

### DISCUSSION THREAD

**Discussion by eldanh** [containerd/nerdctl#975] [Q&A] [answered]
How can I save an image as a tar in docker format (having 'repositories' and 'manifest.json') and not the 'blobs' format?

**Comment by fahedouch** [comment, marked as answer]
==> https://github.com/containerd/nerdctl#whale-nerdctl-save 

`nerdctl save`  implement  both Docker Image Spec v1.2 and OCI Image Spec v1.0.

**Comment by eldanh** [comment]
Hi, I tried and could not generate the proper format.

Tried executing 'nerdctl save' as such:

`nerdctl save XXX --output x.tar`

but am getting this format

> drwxr-xr-x 0/0               0 1970-01-01 02:00 blobs/
> drwxr-xr-x 0/0               0 1970-01-01 02:00 blobs/sha256/
> -r--r--r-- 0/0            1580 1970-01-01 02:00 blobs/sha256/565XXXXXXXXXXXXXXXXXX6ad
> -r--r--r-- 0/0         1750976 1970-01-01 02:00 blobs/sha256/5b0XXXXXXXXXXXXXXXXXX1a9
> -r--r--r-- 0/0            3490 1970-01-01 02:00 blobs/sha256/ca8XXXXXXXXXXXXXXXXXX2ab
> -r--r--r-- 0/0         8651777 1970-01-01 02:00 blobs/sha256/dedXXXXXXXXXXXXXXXXXXe1d
> -rw-r--r-- 0/0             330 1970-01-01 02:00 index.json
> -rw-r--r-- 0/0             632 1970-01-01 02:00 manifest.json
> -r--r--r-- 0/0              30 1970-01-01 02:00 oci-layout
> 

while I need the docker save format as such (from 'docker save')

> drwxr-xr-x 0/0               0 2021-12-19 01:27 d09XXXXXXXXXXXXXXXXXXXb61/
> -rw-r--r-- 0/0               3 2021-12-19 01:27 d09XXXXXXXXXXXXXXXXXXXb61/VERSION
> -rw-r--r-- 0/0             401 2021-12-19 01:27 d09XXXXXXXXXXXXXXXXXXXb61/json
> -rw-r--r-- 0/0         5879808 2021-12-19 01:27 d09XXXXXXXXXXXXXXXXXXXb61/layer.tar
> -rw-r--r-- 0/0             604 1970-01-01 02:00 manifest.json
> -rw-r--r-- 0/0             106 1970-01-01 02:00 repositories

I also tried to create the format with the 'nerdctl build' but same 'blob' format result, as such:

`nerdctl build -t XXX --output type=docker,dest=./a.tar .`

**Reply by AkihiroSuda** [reply]
That is "proper" format covering both Docker Image Spec (`manifest.json`) and OCI Image Spec (`index.json`).
The `blobs` are referred by the both JSONs.
